lib/dnscache.c

```diff
@@ -89,6 +111,15 @@ static gint dns_cache_expire = 3600;
 static gint dns_cache_expire_failed = 60;
 static gint dns_cache_persistent_count = 0;
 static gchar *dns_cache_hosts = NULL;
+pthread_mutex_t lock;
```

Comment:
We should probably get rid of all those TLS variables in favour of using a single pointer to the current DNSCache instance.

That will probably solve the name clashes (the need for the final underscore in the member names) and
also make the code a lot simpler.